### ISSUE TITLE
How to compile CCtools

### ISSUE THREAD

**Issue by djd0723** [macchrome/winchrome#125]
In the cross-compilation resources, I was wondering what commands were used to compile cctools using the cctools-port repo?
https://github.com/macchrome/winchrome/wiki/Ungoogled-Chromium-Cross-Compilation-Resources
https://github.com/tpoechtrager/cctools-port

I am trying to cross-compile chromium for a different version than v117.0.5938.92 and I want to compile cctools from scratch to ensure everything works properly for a different version of XCode. How were you able to do this?

**Comment by macchrome** [comment]
I've been investigating this situation: cctools-port is not necessary for the building of (cross-compiled macOS) Chromium. The Chromium build references 'strip', 'install-name-tool' and 'llvm-otool' (which might not be actually/strictly necessary because it's in third_party/llvm-build/Release+Asserts/bin) are all that is required: symbolic links are sufficient.


Package tool chain: https://chromium.googlesource.com/chromium/src/+/refs/tags/xxx.x.xxx.xx[x]/build/xcode_binaries.yaml

tar xJf /home/chromium/store/xcode_binaries_SDK-xx.x-xxxxxx.tar.xz -C build/mac_files/xcode_binaries

rm -rf build/mac_files/xcode_binaries/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/*

cd  build/mac_files/xcode_binaries/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

ln -s  ../../../../../../../../../third_party/llvm-build/Release+Asserts/bin/llvm-objcopy strip

ln -s  ../../../../../../../../../third_party/llvm-build/Release+Asserts/bin/llvm-objdump llvm-otool [OPTIONAL]

ln -s  ../../../../../../../../../third_party/llvm-build/Release+Asserts/bin/llvm-objcopy install_name_tool

**Confirmation - build of Cromite which is more demanding than ungoogled:**

  13978 pts/0    S      0:00 /bin/sh -c if [ ! -e "obj/chrome/chrome_framework_shared_library/Cromite Framework" -o ! -e "obj/chrome/chrome_framework_shared_library/Cromite Framework.TOC" ] || ../../**third_party/llvm-build/Release+Asserts/bin/llvm-otool** -l "obj/chrome/chrome_framework_shared_library/Cromite Framework" | grep -q LC_REEXPORT_DYLIB ; then TOOL_VERSION=1696987305 ../../build/toolchain/apple/linker_driver.py -Wcrl,driver,../../third_party/llvm-build/Release+Asserts/bin/clang++ -Wcrl,strippath,../../**build/mac_files/xcode_binaries/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip** -Wcrl,installnametoolpath,../../**build/mac_files/xcode_binaries/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/install_name_tool** -shared  -Wcrl,dsym,. -Wcrl,dsymutilpath,../../**tools/clang/dsymutil/bin/dsymutil** -compatibility_version 5993.0.118 -current_version 5993.0.118 -Wl,-install_name,@executable_path/../Frameworks/Cromite\ Framework.framework/Versions/118.0.5993.118/Cromite\ Framework -fuse-ld=lld -Wl,-fatal_warnings -Wl,--icf=all -Wl,--color-diagnostics -Wl,-mllvm,-instcombine-lower-dbg-declare=0 -flto=thin -Wl,--thinlto-jobs=all -Wl,-cache_path_lto,thinlto-cache -Wcrl,object_path_lto -Wl,--thinlto-cache-policy=cache_size=10\%:cache_size_bytes=40g:cache_size_files=100000 -Wl,-mllvm,-import-instr-limit=30 -Wl,-mllvm,-disable-auto-upgrade-debug-info -fwhole-program-vtables -Wl,--strict-auto-link --target=x86_64-apple-macos -no-canonical-prefixes -Wl,-dead_strip -Wl,-no_data_in_code_info -Wl,-no_function_starts -nostdlib++ -isysroot 

In summary, given that the resulting Chromium.app is manually packaged (in a cross-compile environment) the install_name_tool symbolic link is not required. the only sybolic link required is strip. [verified]

CCtools is not required and the the ld64 port is definitely not required (requires a llvm/swift toolchain).
